index.js

```diff
@@ -125,36 +150,53 @@ module.exports = function (options, wp, done) {
       }
     });
 
-    // In watch mode webpack returns a wrapper object so we need to get
-    // the underlying compiler
-    if (options.watch) {
-      compiler = compiler.compiler;
-    }
+    var handleCompiler = function (compiler) {
+      // In watch mode webpack returns a wrapper object so we need to get
+      // the underlying compiler
+      if (options.watch && compiler.compiler) {
+        compiler = compiler.compiler;
+      }
 
-    if (options.progress) {
-      compiler.apply(new ProgressPlugin(function (percentage, msg) {
-        percentage = Math.floor(percentage * 100);
-        msg = percentage + '% ' + msg;
-        if (percentage < 10) msg = ' ' + msg;
-        gutil.log('webpack', msg);
-      }));
-    }
+      if (options.progress) {
+        compiler.apply(new ProgressPlugin(function (percentage, msg) {
+          percentage = Math.floor(percentage * 100);
+          msg = percentage + '% ' + msg;
+          if (percentage < 10) msg = ' ' + msg;
+          gutil.log('webpack', msg);
+        }));
+      }
 
-    var fs = compiler.outputFileSystem = new MemoryFileSystem();
+      var fs = compiler.outputFileSystem = new MemoryFileSystem();
 
-    compiler.plugin('after-emit', function (compilation, callback) {
-      Object.keys(compilation.assets).forEach(function (outname) {
-        if (compilation.assets[outname].emitted) {
-          var file = prepareFile(fs, compiler, outname);
-          self.queue(file);
-        }
+      compiler.plugin('after-emit', function (compilation, callback) {
+        Object.keys(compilation.assets).forEach(function (outname) {
+          if (compilation.assets[outname].emitted) {
+            var file = prepareFile(fs, compiler, outname);
+            self.queue(file);
+          }
+        });
+        callback();
       });
-      callback();
-    });
+    };
+
+    if (Array.isArray(options.config) && options.watch) {
```

Comment:
This should probably be documented - it took a while to figure out that the watch needs to go both in the options.config objects as well as the top level options object.  Additionally, webpack 1.13 does not seem to work with watch turned on - https://github.com/webpack/webpack/pull/2498